### PR TITLE
Added feature to get Facebook page and post insights

### PR DESCRIPTION
**Pull request by ZAINABFATIMA0**
User can login through their Facebook account and connect with their FB pages.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+*.pyc
+__pycache__/
+settings.py
```
Comment (sheralim012) [inline .gitignore]: why add settings.py, this is necessary for project to be functional, right?

File: db.sqlite3
Comment (sheralim012) [inline db.sqlite3:1]: add this to gitignore

File: fb_logging/models.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from django.conf import settings
+from django.db import models
+from social_django.models import UserSocialAuth
+
+class UserProfile(models.Model):
+    user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    access_token = models.CharField(max_length=255, blank=True, null=True)
```
Comment (sheralim012) [inline fb_logging/models.py]: we will be using simplejwt for auth and we wouldn't have to store the token

**Review by sheralim012 — CHANGES_REQUESTED** [review]
we wouldn't be working on templates anymore, remove those

File: example.env [hunk 3]
```diff
@@ -0,0 +1,3 @@
+SOCIAL_AUTH_FACEBOOK_KEY = "your_fcaebook_app_id"
```
Comment (sheralim012) [inline example.env]:
```suggestion
SOCIAL_AUTH_FACEBOOK_KEY='your_fcaebook_app_id'
```

File: insightful/settings.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+from decouple import config
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = config("SECRET_KEY")
```
Comment (sheralim012) [inline insightful/settings.py:23]: this isn't added in example env file

File: insightful/settings.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+LANGUAGE_CODE = "en-us"
+
+TIME_ZONE = "UTC"
+
+USE_I18N = True
+
+USE_TZ = True
+
+ALLOWED_HOSTS = ["localhost", "127.0.0.1"]
```
Comment (sheralim012) [inline insightful/settings.py:115]: there is an other variable with the same name at the start of the file

File: users/helper.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+                data = response.json()
+                if 'data' in data and len(data['data']) > 0:
+                    page_info = data['data'][0]
+                    return page_info.get('access_token'), page_info.get('id')
+                else:
+                    return None
+            else:
+                return None
+        else:
+            return None
```
Comment (sheralim012) [inline users/helper.py]: these else statements aren't doing much and wouldn't make difference if you remove them, also this much nesting should be avoided by early exit

**Review by sheralim012 — CHANGES_REQUESTED** [review]

File: insightful/settings.py (hunk 4 above)
Comment (ZAINABFATIMA0) [reply insightful/settings.py:23]: I  have added this secret key in third line of this file. SECRET_KEY='your_project_secret_key'

**Review by sheralim012 — APPROVED** [review]

**Pull request merged**